### PR TITLE
chore: Add Security policy

### LINKED ISSUE #7608: Magma needs a security policy / process to deal with security flaws
# Magma needs a Security Policy

From time to time Magma will be found to have security related issues where disclosure could put existing Magma deployments at risk.  While the Magma project is firmly committed to maintaining an open and transparent development process, we do require a more private process to handle security issues while fixes are developed to address them.

Currently the only way to report and track issues in Magma is through GitHub Issues.  For certain issues related to serious security flaws the project needs a mechanism to privately report issues and a process that ensures these issues are tracked, addressed at high priority, and ultimately disclosed when fixes are available to be applied.

I propose here the creation of a security policy for the Magma Core Foundation and implementation of the processes required to implement it.

## Best Practices

An informal survey of other Open Source projects, discussion with Linux Foundation Staff & Open Infra Foundation Staff, and review of [The CERT Guide to Coordinated Vulnerability Disclosure](https://resources.sei.cmu.edu/asset_files/SpecialReport/2017_003_001_503340.pdf) suggests that a good security policy have the following properties:

- A confidential reporting mechanism
- A group of project members committed to addressing security issues quickly and confidentially
- Inclusion of the person or entity that reported the issue in the resolution
- High priority (P0) assigned to developing and landing required code changes to close the issue
- A mechanism to track security issues that maintains their confidentiality while a fix is being developed
- Clear and transparent communication of the issue delivered coincidently with the patch to fix it

Examples of Open Source projects found during this survey with excellent security policies include [VS Code](https://github.com/microsoft/vscode/blob/main/SECURITY.md) and [Kata Containers](https://github.com/kata-containers/community/blob/master/VMT/VMT.md).

## Proposal

I propose that the Magma Foundation publish a security policy and creates the processes to implement that policy ASAP.  Github provides a well recognized mechanism to publish security policies through the creation of a SECURITY.md file at the top level.  I will be submitting a draft PR with a proposed security policy for discussion later this week.

The draft policy will be based on the practices found from other projects and advice from LF, OIF and Facebook.  It will strive to satisfy the properties listed above while remaining simple enough for Magma's limited development team to manage successfully.

This policy will require:

- Establishment of a secure e-mail forwarder at `security@magmacore.org`.
- Nomination of and commitment from a small group of Magma's codeowners to serve as a `vulnerability management team`
- Implementation of Github's "security advisory" feature to track issues confidentially and ultimately disclose them when a patch is available

### PR DESCRIPTION
**Pull request by PARitter**
Signed-off-by: Phillip Ritter <pritter@fb.com>

## Summary

Adds github security policy file (SECURITY.md) to document how Magma will handle security vulnerabilities

## Test Plan

Policy statement. Not directly testable.

## Additional Information

Closes #7608

### CODE REVIEW

**Comment by ekowtaylor** [comment]
@electronjoe can you help merge this doc change? cc. @PARitter

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+5. Inform the relevant Magma Codeowners, adding them to the security advisory.
+
+As the fix is being developed, they will then reach out to the reporter to ask them if they would like to be involved and whether they would like to be credited. For credit, the GitHub security advisory UI has a field that allows contributors to be credited.
+
+When the issue is resolved, they will contact the Magma release team and Magma Core Foundation's Outreach Committee to coordinate the publication of the security advisory.
+
+
+For more information on security advisories, see [the GitHub documentation](https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/managing-security-vulnerabilities-in-your-project).
+
+## Security Vulnerabiltiy Team Members
```
[the remark this answers is not in the document]
Comment (electronjoe) [reply SECURITY.md]: Can we fix this @PARitter

**Review by electronjoe — APPROVED** [review]
Minus the one mis-spelling linter finding

**Comment by PARitter** [comment]
Fixed the spelling error...

**Comment by electronjoe** [comment]
Happy to merge whenever @PARitter is ready with ready-2-merge (or perhaps he has permissions himself actually).

**Comment by PARitter** [comment]
I can merge it.  Will have to force-merge since it's failing Py-Lint and CodeQL (amusing!).

**Pull request merged**